services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/Incident.java

```diff
@@ -0,0 +1,415 @@
+
+  /**
+   * {@link ImpactType} critical.
+   */
+  public static final String IMPACT_CRITICAL = "critical";
+
+  /**
+   * {@link ImpactType} major.
+   */
+  public static final String IMPACT_MAJOR = "major";
+
+  /**
+   * {@link ImpactType} minor.
+   */
+  public static final String IMPACT_MINOR = "minor";
+
+  /**
+   * {@link ImpactType} low.
+   */
+  public static final String IMPACT_LOW = "low";
+
+  /**
+   * Impact type.
+   */
+  @Retention(RetentionPolicy.SOURCE)
+  @StringDef({
+    IMPACT_UNKNOWN,
+    IMPACT_CRITICAL,
+    IMPACT_MAJOR,
+    IMPACT_MINOR,
+    IMPACT_LOW
+  })
+  public @interface ImpactType {
+  }
+
+  /**
+   * Unique identifier for incident. It might be the only one <b>non-null</b> filed which meant
+   * that incident started on previous leg and one has an incident with the same <b>id</b>.
+   */
+  @NonNull
+  public abstract String id();
+
+  /**
+   * On of incident type.
+   *
+   * @see IncidentType
+   */
+  @Nullable
+  @IncidentType
+  public abstract String type();
+
+  /**
+   * <b>True</b> if road is closed and no opportunity to go there. <b>False</b> otherwise.
+   */
+  @Nullable
+  public abstract Boolean closed();
+
+  /**
+   * Quantitative descriptor of congestion.
+   */
+  @Nullable
+  public abstract Congestion congestion();
+
+  /**
+   * Human readable description of the incident suitable for showing users.
+   */
+  @Nullable
+  public abstract String description();
+
+  /**
+   * Human readable long description of the incident suitable for showing users.
+   */
+  @Nullable
+  @SerializedName("long_description")
+  public abstract String longDescription();
+
+  /**
+   * Severity level of incident.
+   *
+   * @see ImpactType
+   */
+  @Nullable
+  @ImpactType
+  public abstract String impact();
+
+  /**
+   * Sub-type of the incident.
+   */
+  @Nullable
+  @SerializedName("sub_type")
+  public abstract String subType();
+
+  /**
+   * Sub-type-specific description.
+   */
+  @Nullable
+  @SerializedName("sub_type_description")
+  public abstract String subTypeDescription();
+
+  /**
+   * AlertC codes.
+   *
+   * @see <a href="https://www.iso.org/standard/59231.html">AlertC</a>
+   */
+  @Nullable
+  @SerializedName("alertc_codes")
+  public abstract List<Integer> alertcCodes();
+
+  /**
+   * List of lanes blocked by incident.
+   */
+  @Nullable
+  @SerializedName("lanes_blocked")
+  public abstract List<String> lanesBlocked();
+
+  /**
+   * Incident' geometry index start point.
+   */
+  @Nullable
+  @SerializedName("geometry_index_start")
+  public abstract Integer geometryIndexStart();
+
+  /**
+   * Incident' geometry index end point.
+   */
+  @Nullable
+  @SerializedName("geometry_index_end")
+  public abstract Integer geometryIndexEnd();
+
+  /**
+   * Time the incident was created/updated in ISO8601 format.
+   */
+  @Nullable
+  @SerializedName("creation_time")
+  public abstract String creationTime();
+
+  /**
+   * Start time in ISO8601 format.
```

Comment:
What does the start/end time mean? How is it different from creation time? Would it be worth to document it here?

Reply:
I believe creation_time is when the incident information was created. Start-time and end-time refer to when the incident itself begins and ends respectively. cc @oini

Reply:
I thought by the same way 🤔 seems it might confuse, I'll document it